oauth/client_factory.go

```diff
@@ -28,6 +29,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+var (
+	errUnparseableResponse = errors.New("unable to parse workspace  API response for namespace")
```

Comment:
The name `errUnparseableResponse` doesn't correspond with the actual text of the error message too much, IMHO.

Reply:
that still doesn't correspond to the text of the error though :) The error message is specific about what kind of request causes this error, yet the name of the error suggests that it is generic for all responses. What about `errUnableToParseWorkspaceResponse`?

Reply:
ok